deploy/template.yaml

```diff
@@ -63,32 +81,48 @@ Resources:
               Resource:
                 - !Sub "arn:aws:ssm:${AWS::Region}:${AWS::AccountId}:parameter/${AWS::StackName}/AddressSessionTableName"
                 - !Sub "arn:aws:ssm:${AWS::Region}:${AWS::AccountId}:parameter/${AWS::StackName}/AddressSessionTtl"
-                - !Sub "arn:aws:ssm:${AWS::Region}:${AWS::AccountId}:parameter/${AWS::StackName}/clients/*"
+      Events:
+        RetrieveSession:
+          Type: Api
+          Properties:
+            Path: /session
+            Method: post
+
+  PostcodeLookupFunction:
+    Type: AWS::Serverless::Function
+    Properties:
+      CodeUri: ../lambdas/postcode-lookup/build/distributions/postcode-lookup.zip
+      Handler: uk.gov.di.ipv.cri.address.api.handler.PostcodeLookupHandler::handleRequest
       Environment:
         Variables:
-          POWERTOOLS_METRICS_NAMESPACE: session
-          POWERTOOLS_SERVICE_NAME: session
+          POWERTOOLS_SERVICE_NAME: di-ipv-cri-address-api-postcode-lookup
+      Policies:
+        - AWSLambdaBasicExecutionRole
+        - AWSXrayWriteOnlyAccess
+        - Statement:
+            - Sid: ReadSecretsPolicy
+              Effect: Allow
+              Action:
+                - 'secretsmanager:*'
+              Resource:
+                - !Sub "arn:aws:secretsmanager:${AWS::Region}:${AWS::AccountId}:secret:/${AWS::StackName}/OrdinanceSurveyAPIKey*"
+      Events:
```

Comment:
There is a `x-amazon-apigateway-integration` connecting an  API gateway to this lambda, do we need an `Api` event too?

Reply:
Removed as requested